### PR TITLE
Fix for 3.16.1

### LINKED ISSUE #389: Whoops! Cannot reach System Daemon. Invalid token. ¯\_(ツ)_/¯
Fresh pull of docker image now fails to start with: Whoops! Cannot reach System Daemon. Invalid token. ¯\_(ツ)_/¯

Please help!

```
Waiting for daemon to start up...
Welcome to NordVPN! You can now connect to VPN by using 'nordvpn connect'.
Technology is already set to 'NORDLYNX'.
Connecting to XXX #XXX (XXX.nordvpn.com)
You are connected to XXX #XXX (XXX.nordvpn.com)!
############################################################
IP: XX.X.X.X/32
Unable to access interface: Protocol not supported
Private Key:
＼(＾O＾)／
############################################################
Waiting for daemon to start up...
Whoops! Cannot reach System Daemon.
Invalid token. ¯\_(ツ)_/¯
Waiting for daemon to start up...
Whoops! Cannot reach System Daemon.
Invalid token. ¯\_(ツ)_/¯
Waiting for daemon to start up...
Whoops! Cannot reach System Daemon.
Invalid token. ¯\_(ツ)_/¯
Waiting for daemon to start up...
Whoops! Cannot reach System Daemon.
Invalid token. ¯\_(ツ)_/¯

```

### PR DESCRIPTION
**Pull request by mhhplumber**
fixes #389 

upgraded OS to jammy
upgraded nord version to 3.16.1
fixed s6 permissions for check
added sleep to stop race condition for nordvpnd.sock

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
-FROM ghcr.io/linuxserver/baseimage-ubuntu:bionic
+FROM ghcr.io/linuxserver/baseimage-ubuntu:jammy
```
Comment (mhhplumber) [inline Dockerfile:1]: Upgrading to nord version 3.16.1 causes an error that GLIBC 2.28 is not found, upgrading the ubuntu version fixes

File: rootfs/etc/fix-attrs.d/s6 [hunk 2]
```diff
@@ -0,0 +1 @@
+/etc/services.d/nordvpn/data/check false root:root 0755 0755
```
Comment (mhhplumber) [inline rootfs/etc/fix-attrs.d/s6:1]: I was getting a warning that s6 was getting a permission denied attempting to spawn a child process from `./data/check`
this fixes
Let me know if you want me to change the filename

File: rootfs/usr/bin/nord_login [hunk 3]
```diff
@@ -13,6 +13,7 @@ if ! iptables -L > /dev/null 2>&1; then
     sleep 3600
   done
 fi
+sleep 5
```
Comment (mhhplumber) [inline rootfs/usr/bin/nord_login:16]: Not sure this is the best way to handle this. There is a race condition that causes `nordvpn/nordvpnd.sock` to become inaccessible if you attempt to login too soon.

**Comment by jarlbrak** [comment]
@bubuntux can you check this out? Hoping this can fix the issues I'm having with the token auth...

**Comment by bubuntux** [comment]
merged, thanks

**Comment by mhhplumber** [comment]
I think what most users were experiencing was nordvpnd.sock not existing yet. If it doesn't fix for everyone, can try increasing the sleep or just waiting for the socket

**Comment by mhhplumber** [comment]
> > I think what most users were experiencing was nordvpnd.sock not existing yet. If it doesn't fix for everyone, can try increasing the sleep or just waiting for the socket
> 
> Now I'm getting a new error that is stumping me...
> 
> ```
> FATAL: iptables is not functional. Ensure your container config adds --cap-add=NET_ADMIN,NET_RAW
> --- Due to errors, routing has been disabled ---
> ```
> 
> I have both NET_ADMIN,NET_RAW in my docker compose...

Can you post the lines from your docker compose? this line I don't think works `--cap-add=NET_ADMIN,NET_RAW` because it takes the capabilities as one argument -- It works when you add them separately ie "--cap-add=NET_ADMIN --cap-add=NET_RAW" Docker compose version may make a difference here too

**Comment by jarlbrak** [comment]
Eh, I'm done messing with it. Both this and NordLynx don't work for me at all since the past two weeks. Just switched to SurfShark and Glutun and it worked on the first run...

**Comment by DarkGM** [comment]
> > > I think what most users were experiencing was nordvpnd.sock not existing yet. If it doesn't fix for everyone, can try increasing the sleep or just waiting for the socket
> > 
> > 
> > Now I'm getting a new error that is stumping me...
> > ```
> > FATAL: iptables is not functional. Ensure your container config adds --cap-add=NET_ADMIN,NET_RAW
> > --- Due to errors, routing has been disabled ---
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I have both NET_ADMIN,NET_RAW in my docker compose...
> 
> Can you post the lines from your docker compose? this line I don't think works `--cap-add=NET_ADMIN,NET_RAW` because it takes the capabilities as one argument -- It works when you add them separately ie "--cap-add=NET_ADMIN --cap-add=NET_RAW" Docker compose version may make a difference here too

Since I'm having the same issue I will reply since @jarlbrak is switching to Glutun.
Using portainer Stack deployement.
`
version: "3"
services:
  vpn:
    container_name: nordvpn
    image: ghcr.io/bubuntux/nordvpn
    restart: unless-stopped    
    cap_add:
     - NET_ADMIN               # required
     - NET_RAW                 # required in some cases
    environment:    
      - TOKEN=abc
      - CONNECT=United_States
      - TECHNOLOGY=NordLynx    #NordLynx/OpenVPN
    sysctls:
      - net.ipv6.conf.all.disable_ipv6=1  # Recomended if using ipv4 only   `

**Comment by mhhplumber** [comment]
@DarkGM I need a couple other details to try to recreate the behavior. Could you let me know what your base OS is and the CPU architecture?

**Comment by fraPace** [comment]
Hello @mhhplumber ,

I am having the same problem above.

My docker-compose is the following one:

`  ---
version: "3"
services:
  vpn:
    image: ghcr.io/bubuntux/nordvpn
    container_name: vpn
    cap_add:
      - NET_ADMIN
      - NET_RAW
    environment:
      - TOKEN=${VPN_TOKEN}
      - CONNECT='--group p2p'
      - TECHNOLOGY=NordLynx
      - NETWORK=192.168.1.0/24
      - PUID=1000
      - PGID=100
      - TZ=Europe/Paris
      - DNS=127.0.0.11;1.1.1.1;8.8.8.8; # need to manually specify them to support docker DNS
    ulimits:                    # Recommended for High bandwidth scenarios
      memlock:
        soft: -1
        hard: -1
    restart: always`
    
 I usually use Portainer EE to manage it and it worked fine.
 I also saw that some reported that with Portainer they could have have the cap_add set properly so I have manually deployed it using docker-compose up -d without success.
 
 My docker env is the following: CPU: 4 RAM: 16.6 GB - Standalone 20.10.17-qnap7.

**Comment by mhhplumber** [comment]
> Hello @mhhplumber ,
> 
> 
> 
> I am having the same problem above.
> 
> 
> 
> My docker-compose is the following one:
> 
> 
> 
> version: "3"
> 
> services:
> 
>   vpn:
> 
>     image: ghcr.io/bubuntux/nordvpn
> 
>     container_name: vpn
> 
>     cap_add:
> 
>       - NET_ADMIN
> 
>       - NET_RAW
> 
>     environment:
> 
>       - TOKEN=${VPN_TOKEN}
> 
>       - CONNECT='--group p2p'
> 
>       - TECHNOLOGY=NordLynx
> 
>       - NETWORK=192.168.1.0/24
> 
>       - PUID=1000
> 
>       - PGID=100
> 
>       - TZ=Europe/Paris
> 
>       - DNS=127.0.0.11;1.1.1.1;8.8.8.8; # need to manually specify them to support docker DNS
> 
>     ulimits:                    # Recommended for High bandwidth scenarios
> 
>       memlock:
> 
>         soft: -1
> 
>         hard: -1
> 
>     restart: always
> 
>     
> 
> I usually use Portainer EE to manage it and it worked fine.
> 
> I also saw that some reported that with Portainer they could have have the cap_add set properly so I have manually deployed it using docker-compose up -d without success.
> 
>  
> 
>  My docker env is the following: CPU: 4 RAM: 16.6 GB - Standalone 20.10.17-qnap7.
> 
> 
> 
> 

Thanks for the info! Work has me very busy atm but I will spend some time on this the next few days.

**Comment by handsome1987** [comment]
Hey is there any news on this issue?
I also have the invalid token error on docker on my synology nas.

**Pull request merged**